raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py

```diff
@@ -303,8 +303,8 @@ def test_state_wait_secretrequest_invalid_amount():
         block_number,
     )
 
-    assert len(iteration.events) == 1
-    assert isinstance(iteration.events[0], EventPaymentSentFailed)
+    msg = 'The payment event now is emitted when the lock expires'
+    assert events.must_contain_entry(iteration.events, EventPaymentSentFailed, {}) is None, msg
```

Comment:
I think we should add a `must_not_contain?  function for this use case, it's hard to read like this.

Reply:
Also fine for me, but the I'd change the return type to `bool` then and make the `data` argument optional.

Reply:
I don't follow. I was suggesting renaming `must_contain_entry` to `find_entry` and leave things as is.

do you want me to rename and add a `must_contain_entry` which returns `bool`?

Reply:
My inital feeling was that `assert find_entry(...) is None` is hard to read and it would be better to be able to write `assert not find_entry(...)`.

Now that I think about it, that works with `None`, nvm then. Need more ☕️

Reply:
okay, this will be a largish and unrelated change, I will open another PR for this